.github/workflows/azure-e2e-test.yml

```
@@ -165,13 +202,34 @@ jobs:
     steps:
     - uses: actions/checkout@v3
 
+    - name: Set up Go environment
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.20'
+
+    - name: Restore the configuration created before
+      uses: actions/cache/restore@v3
+      with:
+        path: |
+          install/overlays/azure
+          ${{ env.TEST_PROVISION_FILE }}
+        key: caa-config-${{ github.run_id }}-${{ github.run_attempt }}
```

uses: actions/download-artifact@v3
        with:
          name: homework
```suggestion
    - name: Restore the configuration created before
      uses: actions/download-artifact@v3
      with:
        name: e2e-configuration
```